### ISSUE TITLE
Inference over custom images

### ISSUE THREAD

**Issue by JhEdLp** [ldz666666/RiDDLE#7]
This is great work; however, I noticed that currently the inference script only works with the FFHQ dataset. I am interested in using it to encrypt and decrypt custom images.

Could you please guide me on how to extend the functionality of the script to support custom images? I would appreciate any code snippets or documentation.

**Comment by DongzeLi-CASIA** [comment]
You can get the latent code of an custom image using [e4e](https://github.com/omertov/encoder4editing)

**Comment by JhEdLp** [comment]
Thanks, it works.

**Comment by CeLuigi** [comment]
Hi @JhEdLp,

Were you able to get latent codes for your images?

Because I tried to extract them using the e4e source code, but I got codes with different sizes than those provided by the RiDDLE authors. Specifically, I extracted the latent codes for the CelebA-HQ images with size 30000x18x512, while those given for FFHQ are 70000x14x512. I ran the following script:
```
python3 inference.py --images_dir ./celebA-HQ/1024x1024/ --latents_only ../checkpoints/e4e_ffhq_encode.pt
```

Could you help me, please?

**Comment by JhEdLp** [comment]
Hi @CeLuigi 

To extract the latent codes over my custom images, I use the [coach_test.py](https://github.com/ldz666666/RiDDLE/blob/master/coach_test.py) script provided by the authors, and modify the necessary path roots in the [test_options_pwd.py](https://github.com/ldz666666/RiDDLE/blob/master/mapper/options/test_options_pwd.py) script, to use their dataloader and the e4e model to extract the latent codes.

**Comment by CCC-SUN-sys** [comment]
Hi @JhEdLp ,

I followed your previous suggestion and used the coach_test.py script together with the modified paths in test_options_pwd.py to extract and reconstruct latent codes on my own dataset. The extraction runs without errors, but I am encountering a serious reconstruction issue:

Problem：When I invert my CelebV-HQ images using e4e and then reconstruct them with the StyleGAN2 generator (same as in RiDDLE),the reconstructed faces become stylized — they look cartoon-like / oil-painting-like and do not preserve the original identity.

My Question：Is this behavior expected when using e4e on non-FFHQ domains?
Or is there something wrong in my reconstruction pipeline (e.g., the decoder, latent normalization, latent_avg usage)?

Any guidance would be greatly appreciated!

**Comment by JhEdLp** [comment]
Hi @CCC-SUN-sys,

The first way to check if your current pipeline works well, is to invert some samples from FFH-Q (Training set). You expect to get similar images. 

On the other hand, based on my previous experience, e4e don't work well on non-FFHQ domains.
